src/convert/base.jl

```diff
@@ -264,8 +264,8 @@ sexp(::Type{RClass{:character}},st::AbstractString) = sexp(RClass{:character}, s
 function sexp(::Type{RClass{:character}}, a::AbstractArray{T}) where T<:AbstractString
     ra = protect(allocArray(StrSxp, size(a)...))
     try
-        for i in 1:length(a)
-            ra[i] = a[i]
+        for (i, idx) in enumerate(eachindex(a))
+            ra[i] = a[idx]
```

Comment:
Note that this still assumes 1-based indexing for `ra` since `enumerate` will start with 1.

Reply:
yes, but we create `ra` and it's indeed one based

Reply:
Fair, though we also create `a` and `v` as 1-based arrays in the above `rcopy` methods which means the loop specifications there were safe prior to using `eachindex`. I assumed you were going for minimal assumptions but if not then that's fine.

Reply:
`typeof(ra)` is lacking a `eachindex` method and the fallback calls `keys`; I'm going to take a stab at defining something appopriate